test/req/steps_test.exs

```diff
@@ -300,6 +300,35 @@ defmodule Req.StepsTest do
     assert Req.get!(c.url <> "/gzip").body == "foo"
   end
 
+  @tag :tmp_dir
+  test "output/1: path (compressed)", c do
+    Bypass.expect_once(c.bypass, "GET", "/save_to_path", fn conn ->
+      conn
+      |> Plug.Conn.put_resp_header("content-encoding", "x-gzip")
+      |> Plug.Conn.send_resp(200, :zlib.gzip("to be filed"))
+    end)
+
+    response = Req.get!(c.url <> "/save_to_path", output: c.tmp_dir <> "/saved.txt")
+    assert response.body == ""
+    assert File.read!(c.tmp_dir <> "/saved.txt") == "to be filed"
+  end
+
+  @tag :tmp_dir
+  test "output/1: :remote_name (not compressed)", c do
+    Bypass.expect_once(c.bypass, "GET", "/directory/save_this.txt", fn conn ->
+      Plug.Conn.send_resp(conn, 200, "to be filed")
+    end)
+
+    cwd = File.cwd!()
+    on_exit(fn -> File.cd!(cwd) end)
+    File.cd!(c.tmp_dir)
```

Comment:
unfortunately if we do this, we should make the whole test case `async: false` as sooner or later we'll run into a problem.

I'd do this instead:

```elixir
test "..." do
  response = Req.get!(c.url <> "/directory/save_this.txt", output: :remote_name)
  assert response.body == ""
  assert File.read!("save_this.txt") == "to be filed"
after
  File.rm("save_this.txt")
end
```

Reply:
I thought about that but thought about it wrong; you are right. I didn't realize you could use `after` this way; that will help me elsewhere; thanks!